flex/engines/graph_db/app/builtin/shortest_path_among_three.cc

```diff
@@ -0,0 +1,231 @@
+  std::vector<vid_t> next_nei_indexs_;
+  bool find = false;
+  while (!nei_label_.empty() && !find) {
+    for (long unsigned int i = 0; i < nei_index_.size(); i++) {
+      for (int j = 0; j < vertex_size_; j++) {
+        for (int k = 0; k < edge_size_; k++) {
+          if (schema_.has_edge_label(label_t(nei_label_[i]), label_t(j),
+                                     label_t(k))) {
+            auto outedges = txn.GetOutEdgeIterator(
+                nei_label_[i], nei_index_[i], j,
+                k);  // 1.self_label 2.self_index 3.edge_label 4.nei_label
+            while (outedges.IsValid()) {
+              auto neighbor = outedges.GetNeighbor();
+              int64_t v_id = txn.GetVertexId(j, neighbor).AsInt64();
+              if (visit.find(v_id) == visit.end()) {
+                next_nei_labels_.push_back(j);
+                next_nei_indexs_.push_back(neighbor);
+                visit.insert(v_id);
+                parent[v_id] =
+                    txn.GetVertexId(nei_label_[i], nei_index_[i]).AsInt64();
+                if (v_id == v2_id) {
+                  find = true;
+                  break;
+                }
+              }
+              outedges.Next();
+            }
+          }
+          if (schema_.has_edge_label(label_t(j), label_t(nei_label_[i]),
+                                     label_t(k))) {
+            auto inedges = txn.GetInEdgeIterator(
+                nei_label_[i], nei_index_[i], j,
+                k);  // 1.self_label 2.self_index 3.edge_label 4.nei_label
+            while (inedges.IsValid()) {
+              auto neighbor = inedges.GetNeighbor();
+              int64_t v_id = txn.GetVertexId(j, neighbor).AsInt64();
+              if (visit.find(v_id) == visit.end()) {
+                next_nei_labels_.push_back(j);
+                next_nei_indexs_.push_back(neighbor);
+                visit.insert(v_id);
+                parent[v_id] =
+                    txn.GetVertexId(nei_label_[i], nei_index_[i]).AsInt64();
+                if (v_id == v2_id) {
+                  find = true;
+                  break;
+                }
+              }
+              inedges.Next();
+            }
+          }
+        }
+        if (find)
+          break;
+      }
+      if (find)
+        break;
+    }
+    if (find)
+      break;
+    nei_label_ = next_nei_labels_;
+    nei_index_ = next_nei_indexs_;
+    next_nei_labels_.clear();
+    next_nei_indexs_.clear();
+  }
+  if (find) {
+    int64_t v = v2_id;
+    while (v != -1) {
+      result_.push_back(v);
+      v = parent[v];
+    }
+    std::reverse(result_.begin(), result_.end());
+    return true;
+  } else {
+    return false;
+  }
+}
+
+std::vector<int64_t> ShortestPathAmongThree::ConnectPath(
+    std::vector<int64_t>& path1, std::vector<int64_t>& path2,
+    std::vector<int64_t>& path3) {
+  std::vector<int64_t> TSP;
+  int v1v2size = (int) path1.size();
+  int v2v3size = (int) path2.size();
+  int v1v3size = (int) path3.size();
```

Comment:
Using size_t instead of int

Reply:
This indeed needs to be modified. Thanks for pointing it out.